### PR TITLE
Change trades table show condition (Closes: #2257)

### PR DESCRIPTION
**Pull request by alter-eggo**
# Description
This pr closes https://github.com/balancer-labs/frontend-v2/issues/2257. Now investments are being displayed as trades only in `deep` and `stablePhantom` pools.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

Please provide instructions so we can test. Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

## Visual context

Please provide any relevant visual context for UI changes or additions. This could be static screenshots or a loom screencast.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have requested at least 2 reviews (If the PR is significant enough, use best judgement here)
- [ ] I have commented my code where relevant, particularly in hard-to-understand areas
- [ ] If package-lock.json has changes, it was intentional.
- [ ] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

**Comment by markusbkoch** [comment]
Thank you for this. While it is a step in the right direction, I was also hoping we could improve how we interpret swap information on pools with pre-minted BPT, including deep pools.

There's nothing fundamentally different between the two transactions highlighted in the screenshot below - in both cases the user deposited some WMATIC in the pool and received some BPT in return. 
image

https://polygonscan.com/tx/0x0ba03cdce0fc54e2a80303d4e78087cef59fe12932b6ad4900e89e28ac03f02d
https://polygonscan.com/tx/0xebcfc25467035ddb75af0aadf0d058e9ca57c0f13e54b7a78be1fbe00b6183a7

In other words, we should display swaps that have the pool token as the token in/out in the investments tab as exits/joins.

This might pose some challenges wrt to the "Load More" button, as IIUC the swaps tab only fetches swap entities from the subgraph, and the investment tab only fetches join/exit entities. So I understand if you want to merge this as is and handle the additional complexity that I'm proposing in a separate PR.

**Review by markusbkoch — APPROVED** [review]

**Review by garethfuller — APPROVED** [review]
